packages/core/src/util/token.spec.ts

```diff
@@ -211,15 +211,59 @@ describe("getWebidFromTokenPayload", () => {
     );
   });
 
-  it("returns the WebID it the 'webid' claim exists", async () => {
+  it("returns the WebID and clientID if the 'webid' and 'azp' claims exist", async () => {
     const mockJose = jest.requireMock("jose") as jest.Mocked<typeof Jose>;
     mockJose.createRemoteJWKSet.mockReturnValue(
       jest
         .fn<ReturnType<(typeof Jose)["createRemoteJWKSet"]>>()
         .mockResolvedValue((await mockJwk()).publicKey),
     );
     const jwt = await mockJwt(
-      { webid: "https://some.webid#me" },
+      { webid: "https://some.webid#me", azp: "some client" },
+      "https://some.issuer",
+      "https://some.clientId",
+    );
+    const { webId, clientId } = await getWebidFromTokenPayload(
+      jwt,
+      "https://some.jwks",
+      "https://some.issuer",
+      "https://some.clientId",
+    );
+    expect(webId).toBe("https://some.webid#me");
+    expect(clientId).toBe("some client");
+  });
+
+  it("returns the WebID and clientID if the 'sub' and 'azp' claims exist and 'sub' is IRI-like", async () => {
+    const mockJose = jest.requireMock("jose") as jest.Mocked<typeof Jose>;
+    mockJose.createRemoteJWKSet.mockReturnValue(
+      jest
+        .fn<ReturnType<(typeof Jose)["createRemoteJWKSet"]>>()
+        .mockResolvedValue((await mockJwk()).publicKey),
+    );
+    const jwt = await mockJwt(
+      { sub: "https://some.webid#me", azp: "some client" },
+      "https://some.issuer",
+      "https://some.clientId",
+    );
+    const { webId, clientId } = await getWebidFromTokenPayload(
+      jwt,
+      "https://some.jwks",
+      "https://some.issuer",
+      "https://some.clientId",
+    );
+    expect(webId).toBe("https://some.webid#me");
+    expect(clientId).toBe("some client");
+  });
```

Comment:
Instead of updating the existing tests, I think it would be fine to add a separate test just for the clientId, regardless of the `webId`.